tools/test/qa/tests/stability_test/xrun_injection.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+# This script is used to test the ability of alsa xrun handling. To do this
+# you need to enable the xrun debugging in kernel driver first.
+# +CONFIG_SND_DEBUG=y
+# +CONFIG_SND_DEBUG_VERBOSE=y
+# +CONFIG_SND_PCM_XRUN_DEBUG=y
+
+# Copyright (c) 2019, Intel Corporation
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#   * Neither the name of the Intel Corporation nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author: Keqiao Zhang <keqiao.zhang@linux.intel.com>
+#
+
+FORMAT=s16_le	# sample format
+CHANNEL=2	# test channel number
+FREQUENCY=48000	# sample frequency
+INTERVAL=0.2	# interval time for each xrun injection
+MAXLOOP=10000	# max loop for xrun injection
+
+usage(){
+	echo "Eg: ./xrun_injection.sh pcm0p or \
+	./xrun_injection.sh pcm0p audio_file.wav"
```

Comment:
I think we need to prefix all our scripts as they could end up being in $PATH. i.e. sof-test-<name> so this could be "sof-test-xrun.sh" Btw, you can use $0 as script name for usage printing.